content/blog/2021-07-04-why-prefer-symfony-bundles-over-plugins-in-shopware-projects.md

```diff
@@ -0,0 +1,63 @@
+tags: [shopware]
+---
+
+Every developer which created a Plugin did already created a Symfony Bundle. All Shopware plugins are extending from a Shopware Bundle class which extends from the Symfony Bundle.
+You might now ask yourself when plugins are bundles why did you wrote this blog post? Lets first list the difference of an plugin and the two bundles.
+
+| Feature                                            	| Plugin 	| Shopware Bundle 	| Symfony Bundle 	|
+|----------------------------------------------------	|--------	|-----------------	|----------------	|
+| modifying Shopware                                 	| ✔️      	| ✔️               	| ✔️              	|
+| can have migrations                                	| ✔️      	| ✔️               	| ❌              	|
+| can be a theme                                     	| ✔️      	| ✔️               	| ✔️              	|
+| can be installed, uninstalled (lifecyle)           	| ✔️      	| ❌               	| ❌              	|
+| can be managed by the shop owner in administration 	| ✔️      	| ❌               	| ❌              	|
+
+## The benefits of working with a Bundle
+
+As an agency you don't want that the shop owner can manage your built extensions. The lifecycle of the plugins is mostly intended for store distribution.
+When you make changes on a project, you want that your code is always active regardless of the state of it. This makes Shopware updates also smoother, as the upgrade process runs without plugins.
+This solves theme compile issues on update process
+You don't want to manage your own built extensions for that project too in your deployment together with the other plugins from the store. 
+Also you could store the entire Code in the template of the Shopware 6 project.
+
+## What's about Tasks which happens normally in Install / Update?
+
+When your Bundle extends from Shopware Bundle, your bundle will automaticly register itself to the migration system. So you can write as usual migrations and run them in the cli using the `database:migrate` command.
+Shopware keeps track also of extending the Administration etc. For tasks like creating entities, you could implement an simple command which does it for you. 
+It's also easier to test a command instead a entire plugin lifecycle. 
+
+## Bundle as a Theme?
+
+A bundle can be also a Theme in Shopware. You will need to implement the `ThemeInterface` like [here](https://github.com/shopware/platform/blob/trunk/src/Storefront/Storefront.php#L23) and run later `theme:refresh` in the console.
+
+## How will the Bundle integrated?
+
+The bundle will be integrated like Symfony usual in the `config/packages.php` [file](https://github.com/shopware/production/blob/6.4/config/bundles.php#L19).
+You could integrate your Bundle code into the `src` folder of the template and register a new namespace in the project `composer.json`
```

Comment:
```suggestion
You could integrate your bundle code into the `src` folder of the template and register a new namespace in the project `composer.json`
```

Reply:
Why the template, that doesn't sound intuitive

Reply:
I did mean the production template :D